src/main/java/org/ml_methods_group/algorithm/attributes/AttributesStorage.java

```diff
@@ -0,0 +1,112 @@
+package org.ml_methods_group.algorithm.attributes;
+
+import com.sixrr.metrics.Metric;
+import com.sixrr.metrics.metricModel.MetricsResult;
+import com.sixrr.metrics.metricModel.MetricsRun;
+import org.jetbrains.annotations.NotNull;
+import org.ml_methods_group.algorithm.Algorithm;
+import org.ml_methods_group.algorithm.entity.*;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * One object of this class must be created per {@link Algorithm} execution. This class stores all
+ * relevant code entities (classes, methods, ...) along with their
+ * {@link org.ml_methods_group.algorithm.entity.RelevantProperties} and features extracted
+ * specially for one particular {@link Algorithm} from metrics this algorithm requested through
+ * {@link Algorithm#requiredMetrics}.
+ */
+public class AttributesStorage {
```

Comment:
FeatureStorage?

Reply:
It is supposed that in the future `CodeEntity` and `RelevantProperties` will be separated and `RelevantProperties` will become more flexible like features currently are. With this separation `AttributesStorage` will have to store `RelevantProperties` as well (actually that was initial implementation). That is why it's called `AttributesStorage` where attributes refers to any kind of information that is derived from raw data.